msu/systems/serialization/serialization_types/serialization_data.nut

```diff
@@ -0,0 +1,60 @@
+::MSU.Serialization.Class.SerializationData <- class extends ::MSU.Serialization.Class.ArrayData
+{
+	__MetaData = null;
+
+	constructor( _data = null )
+	{
+		if (_data == null)
+			_data = [];
+		base.constructor(_data);
+		this.__Type = ::MSU.Serialization.DataType.SerializationData;
+		this.__MetaData = ::MSU.Class.MetaDataEmulator();
+	}
+
+	function getData()
+	{
+		return this;
+	}
+
+	function getDataRaw()
+	{
+		return this.__Data;
+	}
+
+	function push( _element )
+	{
+		this.__Data.push(_element);
+		this.__DataArray.push(::MSU.Serialization.__convertValueFromBaseType(_element));
+	}
+
+	function write( _element, _type )
+	{
+		this.__Data.push(_element);
+		this.__DataArray.push(::MSU.Serialization.__convertValueFromGivenType(_element, _type));
+	}
+
+	function serialize( _out )
+	{
+		this.__MetaData.setRealMeta(::MSU.System.Serialization.SerializationMetaData);
```

Comment:
As mentioned above based on @LordMidas's suggestion I now think this should be optionally behind a flag.

Reply:
We decided to not do this - and instead enforce that the metadata is always the correct metadata from the game. If people want custom things, they can store them in custom data containers.